cdap-api-common/src/main/java/io/cdap/cdap/internal/io/SchemaTypeAdapter.java

```diff
@@ -197,24 +202,27 @@ private Schema readObject(JsonReader reader, Map<String, Schema> knownRecords) t
               Full schema corresponding to this RECORD will be put in knownRecords once the fields in the
               RECORD are explored.
             */
-            knownRecords.put(elementName, null);
+            definedTypes.put(elementName, Schema.recordOf(elementName));
```

Comment:
Is it possible to have duplicate element names when using nested structures?

Something like:

```
{
	"rec1": {
		"a": "value"
	},
	"rec2": {
		"a": "othervalue"
	}
}
```

If so, I think these collisions might cause serialization/deserialization issues for us

Reply:
technically yes it is possible because avro has the concept of a namespace (https://avro.apache.org/docs/current/spec.html#names), with namespace + name needing to be unique.

However, our existing logic doesn't handle that so I didn't do anything. This is potentially something that somebody will run into, so I've opened https://cdap.atlassian.net/browse/CDAP-19035.